sw/host/tests/crypto/hmac_kat/src/main.rs

```diff
@@ -102,24 +102,34 @@ fn run_hmac_testcase(
     .send(&*uart)?;
 
     let hmac_tag = CryptotestHmacTag::recv(&*uart, opts.timeout, false)?;
-    let success = if test_case.tag.len() > hmac_tag.tag_len {
-        // If we got a shorter tag back then the test asks for, we can't accept the tag, even if
-        // the beginning bytes match.
-        false
-    } else {
-        // Some of the NIST test cases only specify the beginning bytes of the expected tag, so we
-        // only check up to what the test specifies.
-        test_case.tag[..] == hmac_tag.tag[..test_case.tag.len()]
-    };
-    if test_case.result != success {
-        log::info!(
-            "FAILED test #{}: expected = {}, actual = {}",
-            test_case.test_case_id,
-            test_case.result,
-            success
-        );
+    let mut failed = false;
+    for (mode, actual_tag) in [
+        ("oneshot", hmac_tag.oneshot_tag),
+        ("streaming", hmac_tag.streaming_tag),
+    ]
+    .into_iter()
```

Comment:
Nit: I think Rust will automatically call `into_iter` in `for` loops, so you could remove this.

Reply:
Thanks, turns out this works for any type that implements `IntoIterator`, including arrays.